hedera-mirror-rest/middleware/metricsHandler.js

```
@@ -0,0 +1,52 @@
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 - 2020 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+'use strict';
+
+const config = require('../config');
+const swStats = require('swagger-stats');
+
+// Load your swagger specification
+// const apiSpec = require('./swagger.json');
+
+const metricsHandler = () => {
+  return swStats.getMiddleware({
+    name: process.env.npm_package_name,
+    version: process.env.npm_package_version,
+    authentication: true,
+    onAuthenticate: onMetricsAuthenticate,
+    uriPath: '/swagger',
+    // swaggerSpec: apiSpec,
+    // metricsPrefix: 'mirror-rest-'
+  });
+};
+
+const onMetricsAuthenticate = async (req, username, password) => {
+  return new Promise(function (resolve, reject) {
+    const match = username === config.db.apiUsername && password === config.db.apiPassword;
+    resolve(match);
+  }).catch((err) => {
+    logger.debug(`Auth error: ${err}`);
```

q: Does catching this error here swallow it instead of failing authentication?

---

No it fails. It needs a valid resolve of true to let you on. I'll check about regrowing though

---

Verified regrowing does now swallow the error and auhtenmicationonly passes with a valid match and a resolve on the promise.

---

Added the throw anyway